### ISSUE TITLE
Support environment variables with certain escape characters

### ISSUE THREAD

**Issue by tylerjl** [kbsecret/kbsecret#62]
Apologies for removing the issue template, this isn't a bug per se but a somewhat generic question that I'm not sure how to solve with a PR so I thought I'd confer with you first.

I use kbsecret quite a bit now to store several types of secrets, including some access keys that include characters like plus signs (`+`). The problem I'm running into is that when setting this values, [`environment` records `shellescape` the string](https://github.com/kbsecret/kbsecret/blob/master/lib/kbsecret/record/environment.rb#L20) which ends up with values like the following:

```console
$ kbsecret new env foobar --force -x <<< "VARIABLE:thing+with+plus"
$ kbsecret env foobar
export VARIABLE=thing\+with\+plus
$
```

I know the upstream ruby docs regarding `shellescape` indicate that things should work, but these types of environment values don't work (as far as I've seen) using tools like `aws`/`cli53`/etc.

I'm not sure what the right approach here is, as escaping the shell variable values makes sense, but out-of-the-box trying to store and then retrieve an environment variable with characters like these hand back values that don't work for all cases.

**Comment by woodruffw** [comment]
Yeah, I'm not sure what the best approach is here. KBSecret *needs* to perform some escaping to avoid malicious inputs, and `Shellwords` is the most reliable way to do that (although I have no idea why it's escaping `+`...).

Out of curiosity: does loading the variable with `env` work? E.g.,

```bash
$ kbsecret new env foobar -fxi = <<< "FOO=bar+baz"
$ env $(kbsecret env foobar -n) aws some-command
```

If it doesn't, I'll leave this open for people to take a look at.

**Comment by tylerjl** [comment]
Yep, the behavior occurs when `AWS_SECRET_ACCESS_KEY` contains a `\+` character:

```
$ env $(kbsecret env -n sts_session_token sts_access_key sts_secret_key) aws ec2 describe-instances

An error occurred (AuthFailure) when calling the DescribeInstances operation: AWS was not able to validate the provided access credentials
```

I renewed my STS credentials a few times to get a secret key _without_ a plus sign and the authentication call succeeded. Oddly, I observed that the STS session key also contained pluses, but requests still succeeded when the secret key didn't have pluses but the STS session key _did_, which is super weird. Things also worked if I manually tweaked all env var values to not escape plus signs. Perhaps this is some CLI tooling consuming env var values badly?

One option could be to pass values through `shellescape` and then perform a small regex find/replace for `s/\\+/+/`. It's kind of ugly, but plus signs aren't problematic for unquoted environment variables like the other characters the scrubbing is helping to handle like quotes/spaces/etc.

**Comment by woodruffw** [comment]
> Perhaps this is some CLI tooling consuming env var values badly?

That's what I'm thinking, yeah.

It's also probably a bug in Ruby's Shellwords: https://github.com/ruby/ruby/blob/trunk/lib/shellwords.rb#L150 -- I don't think `+` is a dangerous character from a shell's perspective.

In any case, doing a small find/replace sounds reasonable. I think I'll expose it behind a flag like `-u`, `--unescape-plus` (which you can make one of your default flags for `kbsecret env`, if you'd like),

**Comment by woodruffw** [comment]
I've released 1.1.0, which contains the fix for this. Thanks for reporting!
